tests/test_widgets.py

```diff
@@ -1071,3 +1071,25 @@ def _exec(self, **k):
     )
     assert vals == dict(age=0, name="")
     mock.assert_called_once()
+
+
+def test_literal():
+    from typing import Set
+
+    from typing_extensions import Literal
+
+    @magicgui
+    def f(x: Literal["a", "b", "c"]):
+        ...
+
+    cbox = f.x
+    assert type(cbox) is widgets.ComboBox
+    assert cbox.choices == ("a", "b", "c")
+
+    @magicgui
+    def f(x: Set[Literal["a", "b", "c"]]):
+        ...
+
+    sel = f.x
+    assert type(sel) is widgets.Select
+    assert cbox.choices == ("a", "b", "c")
```

Comment:
```suggestion
    from typing_extensions import Literal, get_args

    Lit = Literal["a", 1, True, None, b'bytes']

    @magicgui
    def f(x: Lit):
        ...

    cbox = f.x
    assert type(cbox) is widgets.ComboBox
    assert cbox.choices == get_args(Lit)

    @magicgui
    def f(x: Set[Lit]):
        ...

    sel = f.x
    assert type(sel) is widgets.Select
    assert sel.choices == get_args(Lit)
```

let's test all the literal types, just for completeness